src/puppetlabs/pcp/client.clj

```diff
@@ -334,12 +269,10 @@
   [params :- ConnectParams handlers :- Handlers]
   (let [{:keys [cert type server user-data]} params
         client (map->Client {:server server
```

Comment:
Client type should also be appended to the server URI to get the correct connection. Seems like we need a test that uses a non-agent type to make sure that works correctly.